src/ApiService/ApiService/Functions/AgentEvents.cs

```diff
@@ -97,13 +97,14 @@ public class AgentEvents {
             // Don’t check reimage_requested, as nodes only send 'init' state once.  If
             // they send 'init' with reimage_requested, it's because the node was reimaged
             // successfully.
+            // Node has to be in state other than Init, otherwise SetState ignore the change, and therefore
```

Comment:
Explanation on why we need a new state added to nodes. And this is bug number two, that would force nodes to go from Free to Done state right away, since InitializedAt is not set, and so the node is considered to be outdated right after it is created. Since we perform the check on timer and LInux is faster to provision than Windows - Linux nodes succeed more often than Windows

Reply:
This makes a lot of sense.